custom_components/petlibro/binary_sensor.py

```diff
@@ -0,0 +1,106 @@
+"""Support for PETLIBRO binary sensors."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from logging import getLogger
+from collections.abc import Callable
+from datetime import datetime
+from functools import cached_property
+from typing import Any, cast
+from typing import Optional
+
+from homeassistant.components.binary_sensor import BinarySensorEntity, BinarySensorEntityDescription, BinarySensorDeviceClass
+from homeassistant.const import UnitOfMass, UnitOfVolume
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .devices import Device
+from .devices.feeders.feeder import Feeder
+from .devices.feeders.granary_feeder import GranaryFeeder
+from .devices.feeders.one_rfid_smart_feeder import OneRFIDSmartFeeder
+from . import PetLibroHubConfigEntry
+from .entity import PetLibroEntity, _DeviceT, PetLibroEntityDescription
+
+
+_LOGGER = getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class PetLibroBinarySensorEntityDescription(BinarySensorEntityDescription, PetLibroEntityDescription[_DeviceT]):
+    """A class that describes device binary sensor entities."""
+
+    device_class_fn: Callable[[_DeviceT], BinarySensorDeviceClass | None] = lambda _: None
+    should_report: Callable[[_DeviceT], bool] = lambda _: True
+    device_class: Optional[BinarySensorDeviceClass] = None
+
+
+class PetLibroBinarySensorEntity(PetLibroEntity[_DeviceT], BinarySensorEntity):  # type: ignore [reportIncompatibleVariableOverride]
+    """PETLIBRO sensor entity."""
+
+    entity_description: PetLibroBinarySensorEntityDescription[_DeviceT]  # type: ignore [reportIncompatibleVariableOverride]
+
+    @cached_property
+    def device_class(self) -> BinarySensorDeviceClass | None:
+        """Return the device class to use in the frontend, if any."""
+        return self.entity_description.device_class
+
+    @property
```

Comment:
The abstract method is a `cached_property` and not a `property`

```suggestion
    @cached_property
```